docs/src/snforge-advanced-features/conditional-compilation.md

```diff
@@ -0,0 +1,80 @@
+# Conditional Compilation
+
+It is possible to build some contracts solely for testing purposes.
+This can be achieved by leveraging [Scarb features](../snforge-advanced-features/fork-testing.md).
+Configuration in `Scarb.toml` is done in the same manner as described in the Scarb documentation.
+Additionally, the `snforge test` command exposes the following flags: [--features](../appendix/snforge/test.md#-f---features-features),
+[--all-features](../appendix/snforge/test.md#--all-features) and [--no-default-features](../appendix/snforge/test.md#--no-default-features).
+
+## Contracts
+
+Firstly, define a contract in the `src` directory with a `#cfg(feature: (...))` attribute:
+
+```rust
+#[starknet::interface]
+trait IMockContract<TContractState> {
+    fn response(self: @TContractState) -> u32;
+}
+
+#[starknet::contract]
+#[cfg(feature: 'snforge_test_only')]
+mod MockContract {
+    #[storage]
+    struct Storage {}
+
+    #[abi(embed_v0)]
+    impl IContractImpl of super::IContract<ContractState> {
+        fn response(self: @ContractState) -> u32 {
+            1
+        }
+    }
+}
+```
+
+Next, create a test with the same attribute:
+
+```rust
+#[test]
+#[cfg(feature: 'snforge_test_only')]
+fn test_mock_contract() {
+    let (contract_address, _) = declare("MockContract")
+        .unwrap()
+        .contract_class()
+        .deploy(@array![])
+        .unwrap();
+    let response_result = IContractDispatcher { contract_address }.response();
+    assert(response_result == 1, '');
+}
+```
```

Comment:
The cfg flag makes the example unclear imo and I think tagging tests with cfg is a bad practise for most users (I can see usecases for that, but most of them will just forget that the test didn't run). I would remove the cfg and comment in the last line of the paragraph that the test will fail if you don't pass the features flag